aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -162,7 +169,19 @@ private String cleanWarehousePath(String path) {
 
   @Override
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
-    return new GlueTableOperations(glue, lockManager, catalogName, awsProperties, fileIO, tableIdentifier);
+    if (lakeFormationEnabled) {
+      S3FileIO s3FileIO = new S3FileIO();
+      Map<String, String> lakeFormationProperties = Maps.newHashMap(this.catalogProperties);
```

Comment:
prefer to use immutable map, `ImmutableMap.<String, String>builder().putAll(...).put(...).build()`